core/src/test/scala/org/apache/spark/scheduler/OutputCommitCoordinatorSuite.scala

```diff
@@ -0,0 +1,177 @@
+
+  val conf = new SparkConf().set("spark.localExecution.enabled", "true")
+
+  var taskScheduler: TaskSchedulerImpl = null
+  var dagScheduler: DAGScheduler = null
+  var dagSchedulerEventProcessLoop: DAGSchedulerEventProcessLoop = null
+  var accum: Accumulator[Int] = null
+  var accumId: Long = 0
+
+  before {
+    sc = new SparkContext("local", "Output Commit Coordinator Suite")
+    accum = sc.accumulator[Int](0)
+    Accumulators.register(accum, true)
+    accumId = accum.id
+
+    taskScheduler = new TaskSchedulerImpl(sc, 4, true) {
+      override def submitTasks(taskSet: TaskSet) {
+        // Instead of submitting a task to some executor, just run the task directly.
+        // Make two attempts. The first may or may not succeed. If the first
+        // succeeds then the second is redundant and should be handled
+        // accordingly by OutputCommitCoordinator. Otherwise the second
+        // should not be blocked from succeeding.
+        execTasks(taskSet, 0)
+        execTasks(taskSet, 1)
+      }
+
+      private def execTasks(taskSet: TaskSet, attemptNumber: Int) {
+        var taskIndex = 0
+        taskSet.tasks.foreach { t =>
+          val tid = newTaskId
+          val taskInfo = new TaskInfo(tid, taskIndex, 0, System.currentTimeMillis, "0",
+            "localhost", TaskLocality.NODE_LOCAL, false)
+          taskIndex += 1
+          // Track the successful commits in an accumulator. However, we can't just invoke
+          // accum += 1 since this unit test circumvents the usual accumulator updating
+          // infrastructure. So just send the accumulator update manually.
+          val accumUpdates = new mutable.HashMap[Long, Any]
+          try {
+            accumUpdates(accumId) = t.run(attemptNumber, attemptNumber)
+            dagSchedulerEventProcessLoop.post(
+              new CompletionEvent(t, Success, 0, accumUpdates, taskInfo, new TaskMetrics))
+          } catch {
+            case e: Throwable =>
+              dagSchedulerEventProcessLoop.post(new CompletionEvent(t, new ExceptionFailure(e,
+                Option.empty[TaskMetrics]), 1, accumUpdates, taskInfo, new TaskMetrics))
+          }
+        }
+      }
+    }
+
+    dagScheduler = new DAGScheduler(sc, taskScheduler)
+    taskScheduler.setDAGScheduler(dagScheduler)
+    sc.dagScheduler = dagScheduler
+    dagSchedulerEventProcessLoop = new DAGSchedulerSingleThreadedProcessLoop(dagScheduler)
+  }
+
+  /**
+   * Function that constructs a SparkHadoopWriter with a mock committer and runs its commit
+   */
+  private class OutputCommittingFunction
+      extends ((TaskContext, Iterator[Int]) => Int) with Serializable {
+
+    def apply(ctxt: TaskContext, it: Iterator[Int]): Int = {
+      val outputCommitter = new FakeOutputCommitter {
+        override def commitTask(taskAttemptContext: TaskAttemptContext) {
+          super.commitTask(taskAttemptContext)
+        }
+      }
+      runCommitWithProvidedCommitter(ctxt, it, outputCommitter)
```

Comment:
why do you need the anonymous subclass of `FakeOutputCommitter`?  isn't this just
`runCommitWithProvidedCommitter(ctxt, it, new FakeOutputCommitter)`?